main/SCsub

```diff
@@ -6,16 +6,13 @@ from platform_methods import run_in_subprocess
 import main_builders
 
 env.main_sources = []
-env.add_source_files(env.main_sources, "*.cpp")
 
 # order matters here.  higher index controller database files write on top of lower index database files
 controller_databases = ["#main/gamecontrollerdb.txt", "#main/gamecontrollerdb_205.txt", "#main/gamecontrollerdb_204.txt", "#main/godotcontrollerdb.txt"]
 
 env.Depends("#main/default_controller_mappings.gen.cpp", controller_databases)
 env.CommandNoCache("#main/default_controller_mappings.gen.cpp", controller_databases, run_in_subprocess(main_builders.make_default_controller_mappings))
 
-env.main_sources.append("#main/default_controller_mappings.gen.cpp")
```

Comment:
Before moving the `*.cpp` globbing down, and when properly using `env.add_source_files`, I would get this warning added by this commit:
```
Object '['default_controller_mappings.gen.javascript.debug.bc']' already included in environment, skipping.
```